main.go

```diff
@@ -43,26 +62,34 @@ func main() {
 	}
 
 	// Create channel with a few assets and get state back
-	channel, err := v2l.CreateChannel(server, channelName, contentTemplatePath, gopDurMS, nrGopsPerSegment,
+	channels, err := v2l.CreateChannels(nofChannels, server, contentTemplatePath, gopDurMS, nrGopsPerSegment,
 		slidingWindowNrGops, futureScheduleNrGops, assetPaths)
 	if err != nil {
 		log.Fatal(err)
 	}
 
 	signalCh := make(chan os.Signal, 1)
 	signal.Notify(signalCh, os.Interrupt)
-	ticker := time.NewTicker(updatePeriodS * time.Second)
+
+	ticker := time.NewTicker(updatePeriodS * time.Second / time.Duration(nofChannels))
+	chIndex := 0
 TickerLoop:
 	for {
 		select {
 		case <-signalCh:
 			log.Printf("Stopping loop\n")
 			break TickerLoop
 		case t := <-ticker.C:
-			err = v2l.UpdateSchedule(server, channel, assetPaths, t)
+			err = channels[chIndex%nofChannels].UpdateSchedule(server, assetPaths, t)
 			if err != nil {
 				log.Fatal(err)
 			}
 		}
+		chIndex++
```

Comment:
I'd suggest that you do the modulo operation right away, so that one does not suspect an error. index should be less than nrOfChannels. From a performance point of view, it is also better to write it as

chIndex++
if chIndex == nrOfChannels {
    chIndex -= nrOfChannels
}

since divisions and mod are very expensive operations.